lib/TH/generic/THTensor.c

```diff
@@ -285,6 +285,61 @@ void THTensor_(resize5d)(THTensor *self, long size0, long size1, long size2, lon
   THTensor_(resizeNd)(self, 5, size, NULL);
 }
 
+void THTensor_(expandNd)(THTensor *tensor, long *sizes, ptrdiff_t ndim) {
+  THArgCheck(tensor->nDimension > 0, 0, "can't expand an empty tensor");
+  long numUnsqueezed = ndim - tensor->nDimension;
+
+  long *expandedSizes = THAlloc(sizeof(long)*ndim);
+  long *expandedStrides = THAlloc(sizeof(long)*ndim);
+
+  for (long i = 0; i < numUnsqueezed; ++i) {
+    expandedSizes[i] = 1;
+    expandedStrides[i] = 0;
```

Comment:
This seems unnecessary - it's overwritten in the loop below. Maybe just fill the sizes in that one too.